apps/dashboard/src/app/checkout/page.tsx

```diff
@@ -0,0 +1,124 @@
+import "../../global.css";
+import { getThirdwebClient } from "@/constants/thirdweb.server";
+import type { Metadata } from "next";
+import {
+  createThirdwebClient,
+  defineChain,
+  getContract,
+  NATIVE_TOKEN_ADDRESS,
+} from "thirdweb";
+import { name, symbol } from "thirdweb/extensions/common";
+import { decimals } from "thirdweb/extensions/erc20";
+import { checksumAddress } from "thirdweb/utils";
+import { CheckoutEmbed } from "./components/client/CheckoutEmbed.client";
+
+const title = "Universal Bridge: Swap, Bridge, and On-Ramp";
+const description =
+  "Swap, bridge, and on-ramp to any EVM chain with thirdweb's Universal Bridge.";
+
+export const metadata: Metadata = {
+  title,
+  description,
+  openGraph: {
+    title,
+    description,
+  },
+};
+
+export default async function RoutesPage({
+  searchParams,
+}: { searchParams: Record<string, string | string[]> }) {
+  const {
+    chainId,
+    recipientAddress,
+    tokenAddress,
+    amount,
+    clientId,
+    redirectUri,
+  } = searchParams;
+
+  if (!chainId || Array.isArray(chainId)) {
+    throw new Error("A single chainId parameter is required.");
+  }
+  if (!recipientAddress || Array.isArray(recipientAddress)) {
+    throw new Error("A single recipientAddress parameter is required.");
+  }
+  if (!tokenAddress || Array.isArray(tokenAddress)) {
+    throw new Error("A single tokenAddress parameter is required.");
+  }
+  if (!amount || Array.isArray(amount)) {
+    throw new Error("An single amount parameter is required.");
+  }
+  if (Array.isArray(clientId)) {
+    throw new Error("A single clientId parameter is required.");
+  }
+  if (Array.isArray(redirectUri)) {
+    throw new Error("A single redirectUri parameter is required.");
+  }
+
+  // Use any provided clientId or use the dashboard client
+  const client =
+    clientId && !Array.isArray(clientId)
+      ? createThirdwebClient({ clientId })
+      : getThirdwebClient(undefined);
+
+  const token = await (async () => {
+    if (
+      checksumAddress(tokenAddress) ===
+        "0x0000000000000000000000000000000000000000" ||
+      checksumAddress(tokenAddress) ===
+        "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
+    ) {
+      return {
+        name: "Ether",
+        symbol: "ETH",
+        address: NATIVE_TOKEN_ADDRESS,
+        decimals: 18,
+      };
+    } else {
+      const tokenContract = getContract({
+        client,
+        // eslint-disable-next-line no-restricted-syntax
+        chain: defineChain(Number(chainId)),
+        address: tokenAddress,
+      });
+      const symbolPromise = symbol({ contract: tokenContract });
+      const namePromise = name({ contract: tokenContract });
```

Comment:
I think we have a util for this already btw, which also does some.caching and handles native token